DSharpPlus.HttpInteraction.AspNetCore/IEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,74 @@
+using System.Buffers;
+using System.Diagnostics.CodeAnalysis;
+using System.Net.Mime;
+using System.Text;
+using DSharpPlus.Net.HttpInteractions;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace DSharpPlus.HttpInteraction.AspNetCore;
+
+public static class IEndpointRouteBuilderExtensions
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <returns></returns>
+    public static RouteHandlerBuilder AddDiscordHttpInteractions(this IEndpointRouteBuilder builder, [StringSyntax("Route")] string url = "/interactions") => builder.MapPost(url, HandleDiscordInteractionAsync);
+
+    private static async Task HandleDiscordInteractionAsync(HttpContext httpContext, DiscordClient client)
+    {
+        if (!httpContext.Request.Headers.TryGetValue(HeaderNames.ContentLength, out StringValues lengthString) 
+            || !int.TryParse(lengthString, out int length))
+        {
+            httpContext.Response.StatusCode = 400;
+            return;
+        }
+                
+        byte[] bodyBuffer = ArrayPool<byte>.Shared.Rent(length);
+        await httpContext.Request.Body.ReadExactlyAsync(bodyBuffer.AsMemory(..length));
+                
+        if (!TryExtractHeaders(httpContext.Request.Headers, out string? timestamp, out string? key))
+        {
+            httpContext.Response.StatusCode = 401;
+            return;
+        }
+
+        if (!DiscordHeaders.VerifySignature(bodyBuffer.AsSpan(..length), timestamp!, key!, client.CurrentApplication.VerifyKey))
+        {
+            httpContext.Response.StatusCode = 401;
+            return;
+        }
+                
+        byte[] result = await client.HandleHttpInteractionAsync(bodyBuffer[..length]);
```

Comment:
Im pretty sure that there must be a way to remove this allocation. Currently i alloc a new array because the rented array can be bigger than the body

Reply:
arrays in .NET don't support slicing, so unless you want to rework the code to work with ArraySegments/{ReadOnly}Sequences this is unfortunately about as good as it gets